elastalert/ruletypes.py

```diff
@@ -160,26 +160,39 @@ class ChangeRule(CompareRule):
 
     def compare(self, event):
         key = hashable(lookup_es_key(event, self.rules['query_key']))
-        val = lookup_es_key(event, self.rules['compare_key'])
-        if not isinstance(val, bool) and not val and self.rules['ignore_null']:
-            return False
-        changed = False
+        values = []
+        elastalert_logger.info(" Inside Change Rule " + str(self.occurrences))
+        for val in self.rules['compare_key'].split(","):
+            lookup_value = lookup_es_key(event, val)
+            values.append(lookup_value)
+        elastalert_logger.info("  " + str(values) + " " + str(values))
+        for val in values:
+            if not isinstance(val, bool) and not val and self.rules['ignore_null']:
+                elastalert_logger.info("Inside Null" + str(val) + " " + str(isinstance(val, bool)))
+                return False
 
+        changed = False
         # If we have seen this key before, compare it to the new value
         if key in self.occurrences:
-            changed = self.occurrences[key] != val
+            for idx, previous_values in enumerate(self.occurrences[key]):
+                if (len(values) > idx):
+                    elastalert_logger.info(" " + str(previous_values) + " " + str(values[idx]))
+                    changed = previous_values != values[idx]
+                    if(changed):
+                        break
             if changed:
-                self.change_map[key] = (self.occurrences[key], val)
-
+                self.change_map[key] = (self.occurrences[key], values)
                 # If using timeframe, only return true if the time delta is < timeframe
                 if key in self.occurrence_time:
                     changed = event[self.rules['timestamp_field']] - self.occurrence_time[key] <= self.rules['timeframe']
 
         # Update the current value and time
-        self.occurrences[key] = val
+        elastalert_logger.info("Setting current value " + str(values))
+        if (len(values) > 0):
+            self.occurrences[key] = values
         if 'timeframe' in self.rules:
             self.occurrence_time[key] = event[self.rules['timestamp_field']]
-
+        elastalert_logger.info("Final Value " + str(changed))
```

Comment:
add more context. Seeing a log line that says `Final Value abcd` doesn't seem very helpful.